api/api.go

```diff
@@ -1,20 +1,68 @@
 package api
 
-import "fmt"
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/go-resty/resty/v2"
+)
+
+const splitioAPIUri = "https://sdk.split.io/api"
 
 // SplitioAPIBinding contains splitioAPIKey
 type SplitioAPIBinding struct {
 	splitioAPIKey string
+	splitioAPIUri string
 }
 
 // NewSplitioAPIBinding returns a new SplitioAPIBinding
-func NewSplitioAPIBinding(splitioAPIKey string) *SplitioAPIBinding {
-	return &SplitioAPIBinding{splitioAPIKey}
+func NewSplitioAPIBinding(apiKey string, apiURL string) *SplitioAPIBinding {
+	if apiURL == "" {
+		apiURL = splitioAPIUri
+	}
+	return &SplitioAPIBinding{apiKey, apiURL}
 }
 
-// HTTPGet will make a GET request to the Split.io SDK API
-func (binding *SplitioAPIBinding) HTTPGet() error {
-	return fmt.Errorf("not implemented")
+// HTTPGet makes a GET request to the Split.io SDK API.
+// path is the path of the HTTP request e.g "/splitChanges"
+// since is an integer used as a query string, will be -1 on the first request
+func (binding *SplitioAPIBinding) httpGet(path string, since int) (map[string]interface{}, error) {
+	client := resty.New()
+	resp, err := client.R().
+		SetPathParams(map[string]string{
+			"path": path,
+		}).
+		SetQueryParams(map[string]string{
+			"since": string(since),
+		}).
+		SetHeaders(map[string]string{
+			"Accept":        "application/json",
+			"Authorization": fmt.Sprintf("Bearer %s", binding.splitioAPIKey),
+		}).
+		Get(fmt.Sprintf("%s/{path}", binding.splitioAPIUri))
+
+	if err != nil {
+		err = fmt.Errorf("Http get request error: %s", err)
+		return map[string]interface{}{}, err
+	}
+
+	if resp.StatusCode() != http.StatusOK {
+		err = fmt.Errorf("Non-OK HTTP status: %s", resp.Status())
+		return map[string]interface{}{}, err
+	}
+
+	var data map[string]interface{}
```

Comment:
Can you use https://godoc.org/github.com/go-resty/resty#Response.Result here?

Reply:
Do you mean not using `resp.Body()` in line 57, or making the variable `data` another type?

Reply:
Re: thanks for the suggestion. yeah I can use `resp.Result()` by adding SetResult() first:
```
client.R().SetResult(map[string]interface{}{})
```
but the problem I encounter is that the unmarshal method will return every number as float, but I want an integer.

or I can do `SetResult(map[string]json.RawMessage{})` but that will still require some data manipulation later on.
Do you think make it json.RawMessage{} will be better?

Reply:
Re: merge this first, but can come back to modify it if that's better later on